### PR TITLE
Develop NOT for production

### PR DESCRIPTION
**Pull request by Shnoulle**
There are some difference between same domain and other domain

- Other domain : webserver are broken : because SSOWat don't redirect to the good domain (this domain don't have /yunohost currently) ynh bug or this install bug ?
- Then activate LDAP for user 1 seems OK , but https://github.com/YunoHost-Apps/limesurvey_ynh/issues/25 broken (LS bug ? mis-feature ?)

=> Currently user 1 must not be a ynh user : solution create a fake one : done here

### CODE REVIEW

File: conf/config.php.j2 [hunk 1]
```diff
@@ -56,7 +56,12 @@ return array(
         // LimeSurvey developers: Set this to 2 to additionally display STRICT PHP error messages and get full access to standard templates
                 'debug'=>0,
                 'debugsql'=>0, // Set this to 1 to enanble sql logging, only active when debug = 2
-				'enableLdap'=>true
+                /* yunohost part */
+                'auth_webserver_user_map' => array( '{{ admin }}'=>'ynh_admin'),// Primary user as ynh admin
```
Comment (Shnoulle) [inline conf/config.php.j2:60]: admin choosen when installing : super-super admin via webserver only https://github.com/YunoHost-Apps/limesurvey_ynh/issues/25

File: conf/config.php.j2 [hunk 2]
```diff
@@ -56,7 +56,12 @@ return array(
...
+                'auth_webserver_autocreate_user'=>1,
```
Comment (Shnoulle) [inline conf/config.php.j2:61]: Fix https://github.com/YunoHost-Apps/limesurvey_ynh/issues/24

File: conf/data.sql.j2 [hunk 3]
```diff
@@ -1,20 +1,14 @@
...
+INSERT INTO `lime_plugins` (`id`, `name`, `active`) VALUES
 (2,'AuditLog',0),
 (3,'oldUrlCompat',0),
 (4,'ExportR',0),
 (5,'Authwebserver',1),
 (6,'extendedStartPage',0),
 (7,'ExportSTATAxml',0),
 (8,'QuickMenu',0),
-(9,'AuthLDAP',0);
+(9,'AuthLDAP',1);
```
Comment (Shnoulle) [inline conf/data.sql.j2:9]: Fix 'another domain' or AuthWebserver not default with another domain (allow ldap)

File: manifest.json [hunk 4]
```diff
@@ -2,7 +2,7 @@
     "name": "LimeSurvey",
     "id": "limesurvey",
     "packaging_format": 1,
-    "version": "2.62.2",
+    "version": "2.62.2.1",
```
Comment (Shnoulle) [inline manifest.json:5]: I wait for 2.62.3 for final release : false version

File: patches/app-0002-301de6e.patch [hunk 5]
```diff
@@ -0,0 +1,87 @@
+From 301de6eb25dda2b9342a006baadaa068c42b115e Mon Sep 17 00:00:00 2001
+From: Denis Chenu <denis@sondages.pro>
+Date: Wed, 15 Feb 2017 01:59:12 +0100
+Subject: [PATCH] [feature][yunohost] Activate a plugin by command - Needed for
+ update
+
+---
+ application/commands/ActivatePluginCommand.php | 70 ++++++++++++++++++++++++++
```
Comment (Shnoulle) [inline patches/app-0002-301de6e.patch:8]: To move DB direct access system to ls system (with a new ynh plugin (i work on it). Think we can manage really more easily ls via ynh after.

File: scripts/_common.sh [hunk 6]
```diff
@@ -161,28 +161,28 @@ ynh_setup_source () {
...
     sudo mkdir -p "$DEST"
-    sudo chown $AS_USER: "$DEST" 
+    sudo chown $AS_USER: "$DEST"
     if [ "$(echo ${SOURCE_FILE##*.})" == "gz" ]; then
         ynh_exec_as "$AS_USER" tar xf $SOURCE_FILE -C "$DEST" --strip-components 1
     elif [ "$(echo ${SOURCE_FILE##*.})" == "bz2" ]; then
         ynh_exec_as "$AS_USER" tar xjf $SOURCE_FILE -C "$DEST" --strip-components 1
     elif [ "$(echo ${SOURCE_FILE##*.})" == "zip" ]; then
         mkdir -p "/tmp/$SOURCE_FILE"
         ynh_exec_as "$AS_USER" unzip -q $SOURCE_FILE -d "/tmp/$SOURCE_FILE"
```
Comment (Shnoulle) [inline scripts/_common.sh:173]: I try with zip : but seems there are an issue here : user don't have the right on `/tmp/$SOURCE_FILE` not ls related and not really needed

File: scripts/_common.sh [hunk 7]
```diff
@@ -161,28 +161,28 @@ ynh_setup_source () {
...
-        ynh_exec_as "$AS_USER" mv "/tmp/$SOURCE_FILE"/./. "$DEST" 
+        ynh_exec_as "$AS_USER" mv "/tmp/$SOURCE_FILE"/./. "$DEST"
         rmdir "$/tmp/$SOURCE_FILE"
     else
         false
     fi
-    
+
     # Apply patches
-    if [ -f ${PKG_DIR}/patches/$SOURCE_ID-*.patch  ]; then
```
Comment (Shnoulle) [inline scripts/_common.sh:181]: Before : have an error shown  here, now allow multiple patch file without bad sentence in command

**Comment by Shnoulle** [comment]
@zamentur que pense tu des modifications ici ? Est ce que je continu dans ce sens sur l'utilisateur numéro 1 ou bien est ce que je revient sur le principe que l'utilisateur 1 reste celui de ynh.

En fait : ici c'est la mise à jour que j'aurais plus de mal à gérer.

Denis

File: scripts/install [hunk 8]
```diff
@@ -34,29 +34,25 @@ ynh_mysql_generate_db "$user" "$app"
...
-
+# Setup^extra DB part (@todo : replace by a plugin system)
+ynh_configure data.sql ./data.sql
+mysql -u $db_user -p$db_pwd $db_user < ./data.sql
 # Set permissions
 ynh_set_default_perm $local_path
 sudo chmod -R u+w $local_path/tmp
 sudo chmod -R u+w $local_path/upload
-sudo chmod -R u+w $local_path/application/config/
+#~ sudo chmod -R u+w $local_path/application/config/
```
Comment (zamentur) [inline scripts/install:52]: Without write options the ynh_admin can't change the configuration of LS.
What was the idea to remove this ?

File: scripts/install [hunk 9]
```diff
@@ -34,29 +34,25 @@ ynh_mysql_generate_db "$user" "$app"
...
 
 sudo yunohost app addaccess $app -u $admin
-ynh_sso_access "/index.php?r=admin,/index.php?r=plugins,/scripts"
```
Comment (zamentur) [inline scripts/install:59]: I don't remember why I have put a protection on /index.php?r=plugins,/scripts
May be it's historical (with the last version of the package)

File: conf/data.sql.j2 (hunk 3 above)
Comment (zamentur) [reply conf/data.sql.j2:9]: OK, I need to test. If I remember, with AuthLDAP the user is obliged to login twice no ?

**Review by zamentur — COMMENTED** [review]
Thanks, this pr is cool !
The upgrade script is missing, but we can wrote it later.
I need to test the LDAP part.

File: conf/config.php.j2 (hunk 1 above)
Comment (zamentur) [reply conf/config.php.j2:60]: Why do we need to call this user ynh_admin (and not {{ admin }} ) ?

**Comment by zamentur** [comment]
tested, works well.
Configuration seems to be updatable.

About ynh_admin, I don't know I am not sure to understand why it's needed, but you have a better knowledge on LS than me.
I think it could be strange for a user to connect with his username and see "ynh_admin" as the user connected !

File: scripts/install (hunk 8 above)
Comment (Shnoulle) [reply scripts/install:52]: Oups, unsure , maybe a bad removing during testing ... but have another idea here : allow to load user-config from another file (in user directory). Must be reviewed.

File: scripts/install (hunk 9 above)
Comment (Shnoulle) [reply scripts/install:59]: /script : really not needed, about plugins : specific : ls have issue here (plugins/index) but some plugin must allow public access to plugins/

File: conf/config.php.j2 (hunk 1 above)
Comment (Shnoulle) [reply conf/config.php.j2:60]: I do some test : if you remove user access (1st user) from LS : then you don't have a user who can set super-admin ... 

The issue with LS is 

1. Only the 1st super-admin (userid==1) can set other super-admin
2. The 1st super-admin (userid==1) can always log by DBAuth
3. The 1st super admin can not be removed (from LS)
4. The 1st super-admin have always all the rights on all ...

Then : if you set 'ynh-user1' to super-admin at start, play a little, via yunohots : remove this admin, set other admin : the new admin don't are really a real super-admin ....

It's really the point where i didn't know how it's best to fix ....

File: conf/data.sql.j2 (hunk 3 above)
Comment (Shnoulle) [reply conf/data.sql.j2:9]: Yes, but there are some issue with Auth webserver if LS are not a subdirectory of ynh primary server. Last time i test it ....

**Comment by Shnoulle** [comment]
Yes : ynh_admin are really the issue here .....

2 options : 

1. The first admin set : 
     - disallow remove rights via ynh (must be fixed)
2. Use a false admin (like here)
     - can be updated in ynh part.

It's the choice, i can do 2, but not sure for 1 (know less ynh than LS ).

And see : https://github.com/YunoHost-Apps/limesurvey_ynh/issues/25 the super-super-admin can not come from LDAP .....

**Comment by Shnoulle** [comment]
PS : i think i improve LS to allow multiple super-super-admin ;)

**Comment by Shnoulle** [comment]
> I think it could be strange for a user to connect with his username and see "ynh_admin" as the user connected !

After more thinking : you're right . Best seems to have a dedicated user for 1st admin (else you can break system if you remove the rights of the first user (tested with old ynh).

I review next week, update to last version, make a multiple download system for other needed plugins .

**Comment by Shnoulle** [comment]
Must review all :(

**Pull request closed without merging**